### PR TITLE
Reduce example stack usage

### PR DESCRIPTION
**Pull request by c1728p9**
Dynamically allocate buffers that would normally be declared on the
stack. This reduces the maximum stack depth by ~2k.

Note - this just reduces stack usage, the main stack can still overflow even with this patch.

### CODE REVIEW

**Comment by adustm** [comment]
This PR is needed to pass tls-client on NUCLEO_F429ZI.

File: tls-client/main.cpp [hunk 1]
```diff
@@ -310,6 +312,7 @@ class HelloHTTPS {
         _error = !(_got200 && _gothello);
 
         _tcpsocket->close();
+        delete buf;
```
Comment (sbutcher-arm) [inline tls-client/main.cpp]: Second call to `delete buf;` which is a double free.

**Comment by sbutcher-arm** [comment]
Pull request is to fix #11.

**Comment by sbutcher-arm** [comment]
Reviewed and +1 from me. 

@yanesca - Could you please test the PR on a board, and then merge it?

File: tls-client/main.cpp [hunk 2]
```diff
@@ -291,6 +292,7 @@ class HelloHTTPS {
                 print_mbedtls_error("mbedtls_ssl_read", ret);
                 onError(_tcpsocket, -1 );
             }
+            delete buf;
```
Comment (andresag01) [inline tls-client/main.cpp]: Shouldnt this be `delete[] buf`?

File: tls-client/main.cpp (hunk 1 above)
Comment (andresag01) [inline tls-client/main.cpp]: Shouldnt this be `delete[] buf;`?

**Comment by c1728p9** [comment]
Thanks for the review @andresag01. I updated the PR to address your comments.

**Comment by yanesca** [comment]
Reviewed, applied @andresag01 's suggestions, tested and merged.

**Pull request closed without merging**